include/TPP/Dialect/Check/CheckOps.td

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CHECK_DIALECT_CHECK_OPS
+#define CHECK_DIALECT_CHECK_OPS
+
+include "mlir/IR/OpBase.td"
+
+def CHECK_Dialect : Dialect {
+  let name = "check";
+  let cppNamespace = "::mlir::check";
+  let summary = [{
+    A dialect implementing test assertions.
+  }];
+}
+
+def CHECK_ExpectTrueOp : Op<CHECK_Dialect, "expect_true"> {
+  let summary = [{Checks that the operand is true}];
+  let description = [{
+    Verifies that the operand contains a true value, which is represented by
+    any non-zero integer.
+
+    Issues a non-fatal failure if the verification fails.
+
+    ```mlir
+    check.expect_true(%arg0) : i1
+    ```
+  }];
+
+  let arguments = (ins AnyI1:$operand);
+
+  let assemblyFormat = "`(` $operand `)` attr-dict `:` type($operand)";
+}
+
+def CHECK_ExpectAlmostEqOp :
```

Comment:
I don't like the word "almost". It doesn't convey what we're trying to do. In the LLVM test suite / benchmarks we use `fp-comp`. We may get into a state that each number isn't even remotely similar to the next, but the overall distribution is, so this is just a special type of comparison.

We'll probably need to compare integers at some point (INT8), so this also can't be called `FPComp`. I don't know a good name, but "almost" isn't it.

Reply:
How about `expect_within_range`?

Reply:
That name seems to suggest there would be a range minimum and maximum that are passed as arguments to the func. Shall we decide on the name later and get the PR reviewed for now?